### PR TITLE
Add rhel suffix to image references

### PR DESCRIPTION
**Pull request by creydr**
Adds the `-rhelX` suffix for the repos pointing to `registry.redhat.com`.

### CODE REVIEW

**Comment by mgencur** [comment]
The tests fail with:
```
./hack/lib/images.bash: line 18: sorhel: command not found
```

**Comment by mgencur** [comment]
@creydr I think you'll need to call the target ` install-tool-sorhel` in all Make targets, similar to install-tool-branch.
What about creating a Make target that will install all tools?
```
install-tools: install-tool-sobranch install-tool-skopeo install-tool-generate install-tool-sorhel
```
And then use the target like this:
```
operator-e2e: install-tools
```

**Review by pierDipi — COMMENTED** [review]
We also need the "reverse" (aka remove suffix) to create https://github.com/openshift-knative/serverless-operator/blob/main/olm-catalog/serverless-operator-index/image_content_source_policy.yaml in https://github.com/openshift-knative/serverless-operator/blob/main/hack/generate/catalog.sh

**Comment by pierDipi** [comment]
I think here we need to remove the suffix: https://github.com/openshift-knative/serverless-operator/blob/8482ae7159b8373c368dc76aa455a05f92f8d5a6/hack/lib/catalogsource.bash#L146-L147

**Comment by creydr** [comment]
> I think here we need to remove the suffix:
> 
> https://github.com/openshift-knative/serverless-operator/blob/8482ae7159b8373c368dc76aa455a05f92f8d5a6/hack/lib/catalogsource.bash#L146-L147
> 
> because now `img` will be: `serverless-openshift-kn-operator-rhel8` (as an example)

Isn't this, what I am doing in 5a477e6c9fc322193227abce8660f8f9ed9680bf

File: Makefile [hunk 1]
```diff
@@ -384,10 +384,12 @@ install-tool-sobranch:
 	GOFLAGS='' go install github.com/openshift-knative/hack/cmd/sobranch@latest
 
 install-tool-skopeo:
-	GOFLAGS='' go install -tags="exclude_graphdriver_btrfs containers_image_openpgp" github.com/containers/skopeo/cmd/skopeo@v1.16.1
+	#GOFLAGS='' go install -tags="exclude_graphdriver_btrfs containers_image_openpgp" github.com/containers/skopeo/cmd/skopeo@v1.16.1
```
Comment (mgencur) [inline Makefile:387]: Probably a typo here?

**Comment by pierDipi** [comment]
Ah nevermind, that's not necessary since we recreate it every time

File: Makefile (hunk 1 above)
Comment (creydr) [reply Makefile]: :facepalm: 
updated

**Comment by mgencur** [comment]
> Ah nevermind, that's not necessary since we recreate it every time
@pierDipi I think you added [ImageContentSourcePolicy](https://github.com/openshift-knative/serverless-operator/blob/main/olm-catalog/serverless-operator-index/image_content_source_policy.yaml) for local developments and demos? Not exactly sure

**Comment by pierDipi** [comment]
yes, I will break demo env for a bit, but that's fine for now

**Comment by mgencur** [comment]
> Ah this file is not auto generated?

I think it actually is auto-generated and the next "make generated-files" PR would fix it. But we can have it here as well.

**Comment by creydr** [comment]
```
13:40:13.811 INFO: Install ImageContentSourcePolicy
error: unable to read image image-registry.openshift-image-registry.svc:5000/openshift-marketplace/serverless-index:latest: Get "https://image-registry.openshift-image-registry.svc:5000/v2/": tls: failed to verify certificate: x509: certificate signed by unknown authority
13:40:14.018 ERROR: 🚨 Error (code: 1) occurred at ./hack/lib/catalogsource.bash:134, with command: oc adm catalog mirror "$index" "$registry_target" --manifests-only=true --to-manifests=iib-manifests/
```

**Comment by creydr** [comment]
`tls: failed to verify certificate: x509: certificate signed by unknown authority`
Not sure it this is a flake

/retest

**Comment by creydr** [comment]
Konflux:
```
Error: initializing source docker://registry.access.redhat.com/ubi8/ubi-minimal:latest: reading manifest latest in registry.access.redhat.com/ubi8/ubi-minimal: received unexpected HTTP status: 502 Bad Gateway
```

/retest

**Comment by creydr** [comment]
`*[2024-10-23T15:40:21Z]NoClusters: No clusters in pool are ready to be claimed`

/retest

**Comment by creydr** [comment]
```
upgrade.go:25: Serverless downgrade failed: knative kafka downgrade failed: knativekafka knative-kafka is not in desired state, got: &{TypeMeta:{Kind:KnativeKafka APIVersion:operator.serverless.openshift.io/v1alpha1} ObjectMeta:{Name:knative-kafka GenerateName: Namespace:knative-eventing SelfLink: UID:aeb87d02-9fa4-4748-b467-cbd394f4a5a6 ResourceVersion:93402 Generation:2 CreationTimestamp:2024-10-24 08:56:25 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[] Annotations:map[kubectl.kubernetes.io/last-applied-configuration:{"apiVersion":"operator.serverless.openshift.io/v1alpha1","kind":"KnativeKafka","metadata":{"annotations":{},"name":"knative-kafka","namespace":"knative-eventing"},"spec":{"broker":{"defaultConfig":{"bootstrapServers":"my-cluster-kafka-bootstrap.kafka:9092"},"enabled":true},"channel":{"bootstrapServers":"my-cluster-kafka-bootstrap.kafka:9092","enabled":true},"sink":{"enabled":true},"source":{"enabled":true}}}
        ] OwnerReferences:[] Finalizers:[knative-kafka-openshift] ManagedFields:[{Manager:knative-operator Operation:Update APIVersion:operator.serverless.openshift.io/v1alpha1 Time:2024-10-24 08:56:25 +0000 UTC FieldsType:FieldsV1 FieldsV1:{"f:metadata":{"f:finalizers":{".":{},"v:\"knative-kafka-openshift\"":{}}},"f:spec":{"f:broker":{"f:defaultConfig":{"f:authSecretName":{}}},"f:channel":{"f:authSecretName":{},"f:authSecretNamespace":{}},"f:high-availability":{".":{},"f:replicas":{}},"f:logging":{".":{},"f:level":{}}}} Subresource:} {Manager:kubectl-client-side-apply Operation:Update APIVersion:operator.serverless.openshift.io/v1alpha1 Time:2024-10-24 08:56:25 +0000 UTC FieldsType:FieldsV1 FieldsV1:{"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}}},"f:spec":{".":{},"f:broker":{".":{},"f:defaultConfig":{".":{},"f:bootstrapServers":{},"f:numPartitions":{},"f:replicationFactor":{}},"f:enabled":{}},"f:channel":{".":{},"f:bootstrapServers":{},"f:enabled":{}},"f:sink":{".":{},"f:enabled":{}},"f:source":{".":{},"f:enabled":{}}}} Subresource:} {Manager:knative-operator Operation:Update APIVersion:operator.serverless.openshift.io/v1alpha1 Time:2024-10-24 09:15:49 +0000 UTC FieldsType:FieldsV1 FieldsV1:{"f:status":{".":{},"f:conditions":{},"f:version":{}}} Subresource:status}]} Spec:{Broker:{Enabled:true DefaultConfig:{BootstrapServers:my-cluster-kafka-bootstrap.kafka:9092 NumPartitions:10 ReplicationFactor:3 AuthSecretName:}} Source:{Enabled:true} Sink:{Enabled:true} Channel:{Enabled:true BootstrapServers:my-cluster-kafka-bootstrap.kafka:9092 AuthSecretNamespace: AuthSecretName:} Config:map[] HighAvailability:0xc00b4edc20 Logging:0xc00c34b900 Workloads:[]} Status:{Status:{ObservedGeneration:0 Conditions:[{Type:DeploymentsAvailable Status:True Severity: LastTransitionTime:{Inner:2024-10-24 09:07:32 +0000 UTC} Reason: Message:} {Type:InstallSucceeded Status:True Severity: LastTransitionTime:{Inner:2024-10-24 08:56:27 +0000 UTC} Reason: Message:} {Type:Ready Status:False Severity: LastTransitionTime:{Inner:2024-10-24 09:15:49 +0000 UTC} Reason:NotReady Message:Waiting on StatefulSets} {Type:StatefulSetsAvailable Status:False Severity: LastTransitionTime:{Inner:2024-10-24 09:15:49 +0000 UTC} Reason:NotReady Message:Waiting on StatefulSets}] Annotations:map[]} Version:1.14}}: context deadline exceeded
```
But checking on the knative-kafka CR, it was ready:
```
/cli/oc -n knative-eventing get knativekafka
NAME            VERSION   READY   REASON
knative-kafka   1.14      True
```

so maybe a timing issue
/test 416-test-upgrade-aws-416

**Comment by mgencur** [comment]
Yeah, a timing issue. I've seen this on other PRs. It takes a while to start the Pods in StatefulSets.

**Pull request merged**